addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -182,21 +182,26 @@ export const editorCommands = {
         // In case the html inserted is all contained in a single root <p> or <li>
         // tag, we take the all content of the <p> or <li> and avoid inserting the
         // <p> or <li>. The same is true for a <pre> inside a <pre>.
-        if (container.childElementCount === 1 && (
-            container.firstChild.nodeName === 'P' ||
-            container.firstChild.nodeName === 'LI' ||
-            container.firstChild.nodeName === 'PRE' && closestElement(startNode, 'pre')
-        )) {
+        const isBlockElement = (node) => isBlock(node) && !['TABLE', 'UL', 'OL', 'DIV'].includes(node.nodeName);
```

Comment:
Rename to `shouldUnwrap` and include the test about `startNode` in it. Basically this:
```js
const shouldUnwrap = (nodeToInsert, insertionTarget) =>
  insertionTarget.textContent !== "" &&
  isBlock(nodeToInsert) &&
  !["TABLE", "UL", "OL", "DIV"].includes(nodeToInsert.nodeName);
```

Reply:
Can we first merge https://www.odoo.com/odoo/project/1695/tasks/3630662?cids=2 and then I rebase this PR and remove the first commit because I am fixing the same issue in other PR.

Reply:
I just reviewed the other PR.